src/lib.rs

```diff
@@ -209,11 +210,16 @@ where
             while let Ok(task) = task_rx.recv_async().await {
                 match task {
                     CompactTask::Freeze => {
-                        if let Err(err) = compactor.check_then_compaction().await {
+                        if let Err(err) = compactor.check_then_compaction(None).await {
                             // todo!();
                             error!("[Compaction Error]: {}", err)
                         }
                     }
+                    CompactTask::Flush(option_tx) => {
```

Comment:
You can write it like this to avoid repeated `errors!`

```rust
while let Ok(task) = task_rx.recv_async().await {
    if let Err(err) = match task {
        CompactTask::Freeze => compactor.check_then_compaction(None).await,
        CompactTask::Flush(option_tx) => {
            compactor.check_then_compaction(option_tx).await
        }
    } {
        error!("[Compaction Error]: {}", err)
    }
}
```